base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -124,6 +125,33 @@ where T: BlockchainBackend + 'static
                 }
                 Ok(NodeCommsResponse::BlockHeaders(block_headers))
             },
+            NodeCommsRequest::FetchHeadersAfter(header_hashes, stopping_hash) => {
```

Comment:
I like the range search, I think the other sync algorithm needs a similar request.

Reply:
I think this request is useful even if the rest of the PR doesn't make it in.